api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1361,30 +1367,52 @@ public void getUserByEmail_shouldGetUserByUsingEmail() {
 		User user = userService.getUserByEmail("hank.williams@gmail.com");
 		assertNotNull("User with email hank.williams@gmail not found in database", user);
 	}
-	/*
-	 * @see UserService#getUserByActivationKey(String)
-	 */
+	
 	@Test
-	public void getUserByActivationKey_shouldGetUserUsingActivationKey() {
-		executeDataSet(XML_FILENAME);
-		String key="XbklsieiskeuztecEnPf";
-		User user = userService.getUserByActivationKey(key);	
-		assertNotNull("User with activationKey ", user); 
-		String[] tokens = user.getActivationKey().split(":");
-		assertEquals(key,tokens[0]);		
-
+	public void setUserActivationKey_shouldCreateUserActivationKey() {
+		User u = new User();
+		u.setPerson(new Person());
+		u.addName(new PersonName("Benjamin", "A", "Wolfe"));
+		u.setUsername("bwolfe");
+		u.getPerson().setGender("M");
+		User createdUser = userService.createUser(u, "Openmr5xy");
+		assertNull(dao.getLoginCredential(createdUser).getActivationKey());
+		assertEquals(createdUser, userService.setUserActivationKey(createdUser));
+		assertNotNull(dao.getLoginCredential(createdUser).getActivationKey());
+	}
+	
+	@Test 
+	public void getUserByActivationKey_shouldGetUserByActivationKey(){
+		User u = new User();
+		u.setPerson(new Person());
+		u.addName(new PersonName("Benjamin", "A", "Wolfe"));
+		u.setUsername("bwolfe");
+		u.getPerson().setGender("M");
+		User createdUser = userService.createUser(u, "Openmr5xy");
+		String key="h4ph0fpNzQCIPSw8plJI";
+		int validTime = 10*60*1000; //equivalent to 10 minutes for token to be valid
+		Long tokenTime = System.currentTimeMillis() + validTime;
+		LoginCredential credentials = dao.getLoginCredential(createdUser);
+		credentials.setActivationKey("b071c88d6d877922e35af2e6a90dd57d37ac61143a03bb986c5f353566f3972a86ce9b2604c31a22dfa467922dcfd54fa7d18b0a7c7648d94ca3d97a88ea2fd0:"+tokenTime);			
+		dao.updateLoginCredential(credentials);
+		assertEquals(createdUser, userService.getUserByActivationKey(key)); 	
 	}
 	
-	/**
-	 * @see UserService#getUserByActivationKey(String)
-	 */
 	@Test
-	public void getUserByActivationKey_shouldReturnNullIfNoUserFoundWithGivenKey() {
-		executeDataSet(XML_FILENAME);
-		String key="XbklsieiskNoMatchKey";
-		User user = userService.getUserByActivationKey(key);			
-		assertNull(user);
+	public void getUserByActivationKey_shouldReturnNullIfTokenTimeExpired(){
```

Comment:
Actually this is the test I was looking for, I see you have it, so ignore my comment